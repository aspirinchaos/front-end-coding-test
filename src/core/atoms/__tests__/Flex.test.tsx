import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Flex } from '..';

describe('Flex atom', () => {
  it('renders correctly', () => {
    const { container } = render(<Flex>Content</Flex>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with default styles', () => {
    const { container } = render(<Flex>Content</Flex>);
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
    expect(container.firstChild).toHaveStyleRule('flex-direction', 'row');
    expect(container.firstChild).toHaveStyleRule('align-items', 'flex-start');
    expect(container.firstChild).toHaveStyleRule(
      'justify-content',
      'flex-start',
    );
    expect(container.firstChild).toHaveStyleRule('flex-wrap', 'nowrap');
    expect(container.firstChild).not.toHaveStyleRule('width', '100%');
  });

  it('renders with props styles', () => {
    const { container } = render(
      <Flex
        direction={'column'}
        alignItems={'center'}
        justifyContent={'center'}
        wrap={'wrap'}
        fullWidth
      >
        Content
      </Flex>,
    );
    expect(container.firstChild).toHaveStyleRule('flex-direction', 'column');
    expect(container.firstChild).toHaveStyleRule('align-items', 'center');
    expect(container.firstChild).toHaveStyleRule('justify-content', 'center');
    expect(container.firstChild).toHaveStyleRule('flex-wrap', 'wrap');
    expect(container.firstChild).toHaveStyleRule('width', '100%');
  });
});
