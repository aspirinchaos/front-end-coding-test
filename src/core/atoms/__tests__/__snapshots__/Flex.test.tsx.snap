// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex atom renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
  class="c0"
>
  Content
</div>
`;
